### ISSUE TITLE
Stuck typing notifications (client edition)

### ISSUE THREAD

**Issue by richvdh** [vector-im/element-web#2456]
Sometimes it seems that typing notifications get stuck on in the _client_ side; they go away if you switch rooms

**Comment by richvdh** [comment]
#2429 was a previous report of this which suggested it only happened in Firefox, but I've also seen it in Chrome 53.0.2785.116.

**Comment by ara4n** [comment]
i haven't seen this since they were fixed serverside.

**Comment by richvdh** [comment]
Nor have I tbh, though it was pretty rare in the first place. I remain to be convinced, but let's be optimistic.

**Comment by richvdh** [comment]
@Kegsay saw this today

**Comment by Kegsay** [comment]
They go away when you swap rooms, as per the original report (person was Erik over federation). Sometimes they do go away on their own (Amandine on matrix.org) when they send another message.

**Comment by richvdh** [comment]
I observed this today, and dug into it.

It appears that `RoomMember.typing` events are not being propagated to the application. `_events` is empty for each `RoomMember` in the room.

I think the problem is that, when we get a gappy sync, we disable the event listeners for the room while we recreate the room state (including the bit in sync.js that listens for `RoomState.newMember` events and adds re-emitters for `RoomMember.*` events).

So basically: gappy syncs break typing notifications.

**Comment by richvdh** [comment]
> Sometimes they do go away on their own (Amandine on matrix.org) when they send another message.

a `RoomMember.typing` event on any room will trigger a state update, which might explain this

**Comment by Kegsay** [comment]
Gappy syncs are sufficiently rare at run-time that I'm surprised that this is the cause. /me shrugs

**Comment by devnoname120** [comment]
I had this happen to me a few times in the past months using desktop client on Linux (vector-web version 0.9.7).

**Comment by dbkr** [comment]
We suspect this may be fixed by https://github.com/matrix-org/matrix-js-sdk/pull/504

**Comment by weeman1337** [comment]
Closing as done because of 

> We suspect this may be fixed by https://github.com/matrix-org/matrix-js-sdk/pull/504
